Lib/typing.py

```diff
@@ -1804,10 +1858,14 @@ def __new__(mcls, name, bases, namespace, /, **kwargs):
                     )
         return super().__new__(mcls, name, bases, namespace, **kwargs)
 
-    def __init__(cls, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        if getattr(cls, "_is_protocol", False):
-            cls.__protocol_attrs__ = _get_protocol_attrs(cls)
+    def __init__(cls, name, bases, namespace, **kwds):
+        super().__init__(name, bases, namespace, **kwds)
+        if getattr(cls, "_is_protocol", False) and cls.__name__ != "Protocol":
+            cls.__protocol_attrs__ = (
+                _get_local_protocol_members(namespace) | _get_parent_members(cls)
+            )
+            # local_attrs =  _get_local_protocol_members(namespace)
+            # cls.__protocol_attrs__ = local_attrs.union(*map(_get_cls_members, bases))
```

Comment:
Not sure which variant is better...

Reply:
The only difference between `_get_cls_members` and  `_get_parent_members` is that the former checks `cls.__mro__[:-1]` and the latter only `cls.__mro__[1:-1]`.

I would expect the `_get_parent_members` to be better for deeply inherited classes, and the `_get_cls_members` to be better for classes with many disjoint parents.